### DISCUSSION TITLE
Jump Conditional Code Erratum

### DISCUSSION THREAD

**Discussion by knweiss** [speed47/spectre-meltdown-checker#568] [Other vulnerabilities / information]
What about the **Jump Conditional Code (JCC) Erratum**?

Some pointers:

- Intel: [Mitigations for Jump Conditional Code Erratum](https://www.intel.com/content/dam/support/us/en/documents/processors/mitigations-jump-conditional-code-erratum.pdf) whitepaper (pdf)
- Red Hat: [Intel November 2019 Microcode Update ](https://access.redhat.com/solutions/2019-microcode-nov) - _"Under a corner case condition, such a sequence will incorrectly attempt to execute code that does not exist, leading to a range of errors from #UD (Undefined Instruction kernel panic) and spurious page faults (which may also panic the system). Red Hat product security has **not ruled out that privilege escalation is possible using this bug**."_
- Phoronix: [JCC article collection](https://www.phoronix.com/scan.php?page=search&q=JCC%20Erratum) - Great resource with lots of information and tests on the topic.

The **executive summary** is that the latest Intel CPU microcodes will fix this erratum at the cost of performance. Some of the performance loss can be mitigated by changing the code generator of LLVM and GCC and the assemblers. However, this comes with a cost for CPUs **not** affected by this erratum (e.g. AMD). See e.g. [this very interesting discussion of the LLVM developers](https://reviews.llvm.org/D70157#1747640).

What could `spectre-meltdown-checker` offer?

- Report the affected CPUs and the status of the (mitigated) microcode levels
- Show if the kernel was compiled with the conditional jump compiler mitigations to prevent performance loss

Please notice that e.g. the proposed LLVM fix is still in review and not released yet.

**Comment by speed47** [comment]
Agreed, but as it's been somehow a long time since last release, I want to properly tag a new release before adding more stuff.

I wanted at least MCEPSC and TAA to make it to next release, both are now in master, I'll wait a few days to potentially fix any bug on those that might still be reported, before releasing.
Then go on adding JCC and maybe adding some other vulns that have no kernel or microcode mitigation per se but are still interesting to get detected, just so that people know their CPU is vulnerable (maybe only enabled with --paranoid or such however, to avoid freaking out users).

**Comment by speed47** [comment]
Added to the ["unsupported" cve list](https://github.com/speed47/spectre-meltdown-checker/blob/test/dist/doc/UNSUPPORTED_CVE_LIST.md#no-cve--jump-conditional-code-jcc-erratum) as there's nothing for the tool to check for. Issue moved to discussion so it stays visible.
